Documentation/cross-platform-performance-and-eventing.md

```diff
@@ -0,0 +1,287 @@
+- Machine-wide or process-wide.  No process attach required.
+- Collection of owned processes without elevated permissions.
+- Events can be tagged with context such as PID, TID.
+- Out-of-tree but binaries available for many common distros.
+- Data stored in Common Trace Format – designed for interop.
+
+#### Cons
+
+- No hardware counter support (CPU samples, context switches, etc.)
+- No built-in callstack collection.
+
+### SystemTap
+
+#### Pros
+
+- Supports User-mode static tracing including call stacks.
+- Static tracing does not require pre-registration of the event or payload definition when the app starts, which makes EventSource support simple.
+- Out-of-tree but binaries available for many common distros.
+
+#### Cons
+
+- Complex kernel module is generated and compiled on-the-fly based on the tracing script.
+- Static tracing includes a fixed set of static tracing APIs with limited overloads (e.g. int, string).  Can’t consider it strongly typed tracing.
+- User-mode stack trace support requires debug information to support unwinding.  No story for JIT compiled code.  
+- User-mode stack traces are only supported on x86 and x64.
+- Data is stored as unstructured text.
+
+### DTrace4Linux
+
+#### Pros
+
+- Would allow for tracing code and collection script re-use across Linux and Mac.
+
+#### Cons
+
+- Source only – no binary redist.
+- Small subset of actual DTrace functionality.
+- One person’s work rather than many contributions from the community.
+
+### FTrace
+
+#### Pros
+
+- High performance function tracer.
+- In-tree: Basically available for almost every distro.
+
+#### Cons
+
+- Tracing in kernel-mode only.
+- No performance data capture.
+
+### Extended Berkeley Packet Filter (eBPF)
+
+#### Pros
+
+- Should support user-mode static tracing.
+- Possible integration with perf_event.
+
+#### Cons
+
+- Not currently available – currently being integrated into the kernel.
+- Final featureset not clear yet.
+
+## Infrastructure Bring-Up Action Items
+
+- Investigate: Determine if clock skew across the trace files will be an issue.
+- Investigate: Are traces portable, or do they have to be opened on collection machine?
+- Investigate: Do we need rundown or can we use /tmp/perf-$pid.map?  How does process/module rundown work?
+- Implement: Enable frame pointers on JIT compiled code and helpers to allow stacks to be walked.  (PR # 468)
+- Implement: Stack walking using existing stackwalker (libunwind and managed code).
+- Implement: JIT/NGEN call frame resolution - /tmp/perf-$pid.map
+- Implement: Trace collection tool.
+	- Responsible for handling all of the complexity.
+	- Start and stop tracing based on information requested.
+	- **OPEN ISSUE:** Handle rundown if required.
+	- Collect any information needed for off-box viewing (e.g. /tmp/perf-$pid.map).
+	- Compress into one archive that can be copied around easily.
+- Implement: Viewing of data in PerfView
+
+# Mac OS
```

Comment:
It's actually "OS X". You don't actually need the "Mac" part. Mac is the hardware. Same split as iPhone and iOS.